pgjdbc/src/test/java/org/postgresql/test/jdbc2/PreparedStatementTest.java

```diff
@@ -1011,4 +1012,33 @@ public void testStatementDescribe() throws SQLException {
     pstmt.close();
   }
 
+  @Test
+  public void testSetTimeAndTimestamp() throws SQLException {
+    PreparedStatement ps = con.prepareStatement("SELECT ?::timestamp");
+    try {
+      Timestamp ts = new Timestamp(1474997614836L);
+      for (int i = 0; i < 3; ++i) {
```

Comment:
Because PreparedStatement is not cached after first executions. But after 5-th executions (which happens on 3-rd iteration) it become cached.